src/main/scala/is/hail/annotations/CodeOrdering.scala

```diff
@@ -30,6 +30,22 @@ case class CodeOrdering(t: Type, missingGreatest: Boolean) {
   private[this] def doubleCompare(v1: Code[Double], v2: Code[Double]): Code[Int] =
     Code.invokeStatic[java.lang.Double, Double, Double, Int]("compare", v1, v2)
 
+  def compare(mb: EmitMethodBuilder, v1: Code[_], v2: Code[_]): Code[Int] = {
+    val trep = t match {
+      case tc: ComplexType => tc.representation
+      case _ => t
+    }
+    trep match {
```

Comment:
actually this is wrong because `TLocus` and `TInterval` are ComplexTypes, urgh. I actually just wanted to use the `TInt32` comparison for `TCall`, I think. All the other complex types I see have their own ordering.

Reply:
You might just duplicate the int compare for TCall.  It's a one-liner and makes its comparison nice and explicit.

Reply:
yeah, will do. I somehow thought there were more complex types that needed to be handled this way, but I think it's just TCall.